src/abstractops.cpp

```diff
@@ -90,12 +110,17 @@ AbsFpEncoding &getFpEncoding(mlir::Type ty) {
   llvm_unreachable("Unknown type");
 }
 
-
 AbsFpEncoding::AbsFpEncoding(const llvm::fltSemantics &semantics,
-                             unsigned valuebits, string &&fn_suffix)
+      unsigned limitbits, unsigned precbits, unsigned valuebits,
+      AbsFpEncoding* larger_fpty_enc, std::string &&fn_suffix)
      :semantics(semantics), fn_suffix(move(fn_suffix)) {
   assert(valuebits > 0);
-  value_bv_bits = valuebits;
+  // BWs for casting
+  limit_bv_bits = limitbits;
+  prec_bv_bits = precbits;
+  this->larger_fpty_enc = larger_fpty_enc;
+
+  value_bv_bits = limitbits + precbits + valuebits;
```

Comment:
What about keeping
`value_bv_bits = valuebits;`
and using `limitbits`/`precbits` in other places differently?

Reply:
I think I understood what it means.. nevermind